src/DynamoCoreWpf/ViewModels/Core/CustomColorPickerViewModel.cs

```diff
@@ -0,0 +1,218 @@
+        {
+            get
+            {
+                return isColorItemSelected;
+            }
+            set
+            {
+                isColorItemSelected = value;
+                OnPropertyChanged(nameof(IsColorItemSelected));
+            }
+        }
+
+        public CustomColorItem(Color? color, string name) : base(color, name)
+        {
+
+        }
+
+        // Create the OnPropertyChanged method to raise the event
+        // The calling member's name will be used as the parameter.
+        protected void OnPropertyChanged([CallerMemberName] string name = null)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
+        }
+    }
+
+    internal class CustomColorPickerViewModel : NotificationObject
+    {
+        private Color? colorPickerFinalSelectedColor;
+        private ObservableCollection<CustomColorItem> basicColors;
+        private ObservableCollection<CustomColorItem> customColors;
+
+
+        /// <summary>
+        /// Color Selected in the ColorPicker
+        /// </summary>
+        public Color? ColorPickerFinalSelectedColor
+        {
+            get
+            {
+                return colorPickerFinalSelectedColor;
+            }
+            set
+            {
+                colorPickerFinalSelectedColor = value;
+                RaisePropertyChanged(nameof(ColorPickerFinalSelectedColor));
+            }
+        }      
+
+        /// <summary>
+        /// List of Basic Colors that will be displayed in the CustomColorPicker
+        /// </summary>
+        public ObservableCollection<CustomColorItem> BasicColors
+        {
+            get
+            {
+                return basicColors;
+            }
+            set
+            {
+                basicColors = value;
+                RaisePropertyChanged(nameof(BasicColors));
+            }
+        }
+
+        /// <summary>
+        /// List of Custom Colors that will be displayed in the CustomColorPicker
+        /// </summary>
+        public ObservableCollection<CustomColorItem> CustomColors
+        {
+            get
+            {
+                return customColors;
+            }
+            set
+            {
+                customColors = value;
+                RaisePropertyChanged(nameof(CustomColors));
+            }
+        }
+
+        public CustomColorPickerViewModel()
+        {
+            BasicColors = new ObservableCollection<CustomColorItem>();
+            BasicColors = CreateBasicColorsCollection();
+
+            CustomColors = new ObservableCollection<CustomColorItem>();
+            CustomColors = CreateCustomColorsCollection();
+        }
+
+        private static ObservableCollection<CustomColorItem> CreateBasicColorsCollection()
+        {
+
+            ObservableCollection<CustomColorItem> observableCollection = new ObservableCollection<CustomColorItem>();
+
+            List<(int R, int G, int B)> colors = new List<(int, int, int)>()
```

Comment:
Would you add a comment about how the following colors are defined or where (link) to find the definition?

Reply:
@QilongTang extra comments added (let me know if is enough) in the next commit: https://github.com/DynamoDS/Dynamo/pull/13794/commits/a49d970a769a23938d84abc67836a231e4de443d